helix-core/src/match_brackets.rs

```diff
@@ -161,18 +207,72 @@ fn surrounding_bytes(doc: &Rope, node: &Node) -> Option<(usize, usize)> {
     Some((start_byte, end_byte))
 }
 
+/// Checks if `node` or its siblings (at most MATCH_LIMIT nodes) is the specified openiing char
+///
+/// # Returns
+///
+/// The position of the found node or `None` otherwise
+fn find_open_pair(doc: RopeSlice, node: Option<Node>, open: char) -> Option<usize> {
+    iter::successors(node, |node| node.prev_sibling())
+        .take(MATCH_LIMIT)
+        .find_map(|node| {
+            let (pos, c) = as_char(doc, &node)?;
+            (c == open).then_some(pos)
+        })
+}
+
+/// Tests if this node is a pair close char and returns the expected open char
+fn as_close_pair(doc: RopeSlice, node: &Node) -> Option<char> {
+    let close = as_char(doc, node)?.1;
+    PAIRS
+        .iter()
+        .find_map(|&(open, close_)| (close_ == close).then_some(open))
+}
+
+/// Checks if `node` or its siblings (at most MATCH_LIMIT nodes) is the specified closing char
+///
+/// # Returns
+///
+/// The position of the found node or `None` otherwise
+fn find_close_pair(doc: RopeSlice, node: Option<Node>, close: char) -> Option<usize> {
+    iter::successors(node, |node| node.next_sibling())
+        .take(MATCH_LIMIT)
+        .find_map(|node| {
+            let (pos, c) = as_char(doc, &node)?;
+            (c == close).then_some(pos)
+        })
+}
```

Comment:
On a similar line, this implementation is very similar to `find_open_pair` except for the closure going to the previous/next sibling. Maybe we can write it once and accept the closure instead.

Reply:
I found a nice way to abstract their after all that doesn't require a closure

Reply:
Less code to worry about :)